### DISCUSSION TITLE
HTML canvas rendering poorly on output

### DISCUSSION THREAD

**Discussion by abine-hunter-berge** [gotenberg/gotenberg#678] [Q&A]
Canvas rendering (using a variety of charting libraries) is rendering poorly with Gotenberg. Here is an example:

<img width="495" alt="Screen Shot 2023-08-28 at 1 24 00 PM" src="https://github.com/gotenberg/gotenberg/assets/133925263/c6143150-4c62-46df-844a-bd511357b708">

1. Is there a way to configure the viewport & device scale factor like with Puppeteer (`page.setViewport({ width: 1200, height: 800, deviceScaleFactor: 6.0 })`)?
2. Is there a better way to render canvas elements using Gotenberg? 
3. Are there any examples for chart/canvas rendering using Gotenberg? 

I have css set to size: 210mm 297mm (300dpi). Here is the HTML canvas:
```
<div style="position: relative; display: block; width: 100%;">
    <canvas id="chart-active-members" width="1000" height="500"
        style="width: 100%; height: 70mm;"></canvas>
</div>
```

And just for reference, here is an image of Puppeteer rendered PDF (with and without viewport settings it renders much better than Gotenberg):
<img width="828" alt="Screen Shot 2023-08-28 at 1 33 55 PM" src="https://github.com/gotenberg/gotenberg/assets/133925263/6d2d191b-7937-4402-8aea-1fbb5ec44587">

Thank you!

**Comment by gulien** [comment]
Hello @abine-hunter-berge 

Last time I’ve checked, viewport does not work for PDF rendering, only images.

However, the `scale` form fields is supported (see https://gotenberg.dev/docs/routes#page-properties-chromium).

You may also want to play with https://gotenberg.dev/docs/routes#wait-before-rendering.

**Comment by Intrepidd** [comment]
I'm seeing the same issue. @abine-hunter-berge did you manage to fix it in the end ? 
The scale field just zooms the page
